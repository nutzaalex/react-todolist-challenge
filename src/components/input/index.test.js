import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Input from './';

describe('Input', () => {
    it('matches snapshot', () => {
        const comp = renderer.create(
            <Input
                name="item"
                value="new todo"
                placeholder="New Item..."
                onChange={jest.fn()}
            />
        );
        expect(comp.toJSON()).toMatchSnapshot();
    });
    it('calls onChange', () => {
        const onChangeSpy = jest.fn();
        const comp = shallow(
            <Input
                name="item"
                value="todo"
                placeholder="New Item..."
                onChange={onChangeSpy}
            />
        );
        const newValue = 'new todo';

        expect(onChangeSpy).not.toHaveBeenCalled();
        comp.find('input').simulate('change', { target: { value: newValue } });
        expect(onChangeSpy).toHaveBeenCalledTimes(1);
        expect(onChangeSpy).toHaveBeenCalledWith(newValue);
    });
});