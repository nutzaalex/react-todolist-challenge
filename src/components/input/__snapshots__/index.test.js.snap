// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<input
  className={undefined}
  name="item"
  onChange={[Function]}
  placeholder="New Item..."
  type="text"
  value="new todo"
/>
`;
